buildSrc/src/main/groovy/org/elasticsearch/gradle/doc/RestTestsFromSnippetsTask.groovy

```diff
@@ -104,7 +104,7 @@ public class RestTestsFromSnippetsTask extends SnippetsTask {
      * format of the response is incompatible i.e. it is not a JSON object.
      */
     static shouldAddShardFailureCheck(String path) {
-        return path.startsWith('_cat') == false &&  path.startsWith('_ml/datafeeds/') == false
+        return path.startsWith('_cat') == false && path.startsWith('_ml/datafeeds/') == false
```

Comment:
why are the changes to the groovy files needed?

Reply:
For the docs tests where we fire forget follower requests, it is entirely too hard to set up a request that is not going to fail. For example, to set up a request that is not going to fail, we have to know the index UUID of the follower. Obtaining that in a docs test is not possible to my knowledge, we do not have a way to set a value to use in a later docs test? Maybe I am missing something?

However, the docs infrastructure automatically adds `is_false: _shards.failures` to *every* docs test. Therefore, I had to add a hook that enables us to say skip that check for these tests where we expect failures.

Again, if there's a way to avoid the failures that I am not seeing, please let me know!

Reply:
Got it, thanks for the explanation!